README.md

```diff
@@ -1,17 +1,33 @@
 # Drawn Together
 
+# Pre-requisites
+
+- Docker
+- nvm (optional)
+- node v21
+
 # Installation
 
 1. Clone repository
-  * `git clone https://github.com/icedtrees/drawn-together`
-2. Install Docker
+   1. `git clone https://github.com/icedtrees/drawn-together`
+2. `nvm use`
+3. `npm install` (optional - this will be done in the Docker container)
```

Comment:
using docker and also nvm feels a bit superfluous

is there like a way to run a docker command to npm install instead of having to install nvm separately?

or actually what's the benefit of running this step beforehand? is it mainly to move the work outside of the first docker compose?

Reply:
I should've marked `nvm use` as optional too. There's no big reason or benefit to running it outside, it's just that if you wanted to run commands locally, you should use `nvm use` first.

The biggest reason to run commands locally would be when you're adding packages. You can use `docker run -it node:21 npm install --save <package name>` but it's kinda verbose and slow.

The reality is that doing all local dev within docker containers is somewhat uncommon. It does seem superfluous to use both host-npm and Docker, but it also seems most pragmatic to support a hybrid approach. In fact, I was thinking of moving our esbuild steps into npm scripts so that you _could_ run something like `npm build && npm dev`, which is the common way to run things.

Reply:
that makes sense to me. we have a middle ground where we have the right version of node when running commands locally but maybe the operating system or other dependencies might be different

Reply:
>  In fact, I was thinking of moving our esbuild steps into npm scripts so that you could run something like npm build && npm dev, which is the common way to run things.

i was thinking about this idea and realised it also fixes some of the windows mount slowness issues i had been running into, so that's an additional benefit of it.

it seems like we sacrifice dev/prod consistently for the sake of developer productivity, which is kind of cool